src/colorize.cr

```diff
@@ -218,9 +218,9 @@ module Colorize
   @[Flags]
   enum Mode
     # Makes the text bold.
-    Bold = 1
+    Bright
```

Comment:
This doesn't seem right.

Reply:
This is the behavior when `Bright` and `Bold` are `1`:
```cr
"hello".colorize.bright.bright.inspect # => "\e[1;1mhello\e[0m"
"hello".colorize.bright.bold.inspect # => "\e[1;1mhello\e[0m"
"hello".colorize.bold.inspect # => "\e[1;1mhello\e[0m"
"hello".colorize.bright.inspect # => "\e[1;1mhello\e[0m"
```

And this is the behavior after this commit, without them:
```cr
"hello".colorize.bright.bright.inspect # => "\e[1mhello\e[0m"
"hello".colorize.bright.bold.inspect # => "\e[1;1mhello\e[0m"
"hello".colorize.bold.inspect # => "\e[1mhello\e[0m"
"hello".colorize.bright.inspect # => "\e[1mhello\e[0m"
```
The behavior afterwards is better because many useless `;1`s are omitted and this decreases the terminal ANSI escape code parsing time. There is no visual difference between `"\e[1mhello\e[0m"` and `"\e[1;1mhello\e[0m"`.

Reply:
Also, I've found a disadvantage of no longer using macros and the private `MODE_*` constants. 
This is the behavior without this PR:
```cr
"hello".colorize.bright.bright.to_s.inspect # => "\e[1mhello\e[0m"
"hello".colorize.bright.bold.to_s.inspect # => "\e[1mhello\e[0m"
"hello".colorize.bold.to_s.inspect # => "\e[1mhello\e[0m"
"hello".colorize.bright.to_s.inspect # => "\e[1mhello\e[0m"
```
Here, all unnecessary `;1`s are omitted, the perfect behavior.
But with this PR and using this long run time `case` rather than macros, `"hello".colorize.bright.bold.inspect` returns `"\e[1;1mhello\e[0m"` (as you can also see in the other comment in the second code block).

So maybe it would be better afterall to revert to the private `MODE_*` constants since it has more disadvantages than advantages (slower ANSI escape code generation because of run time `case` and what I mentioned above).

Reply:
The main problem is Bright and Bold are the same, the API suggests they aren't.
For example one can choose bright, and in its terminal it prints a bold text or vice-versa.

Either we choose to call it `Bright` or `Bold`, but not both, and we mention that the output may vary across terminals.
I'd vote to keep `Bold`, because both the official [VT100 docs](https://vt100.net/docs/tp83/chapter12.html) and Wikipedia use this therm.

Reply:
I don't really agree. When we keep `Bold` then it wouldn't make sense when you get bright text, when we keep `Bright`, it wouldn't make sense when you get bold text.

I will revert to not using a `case` now. It doesn't really have advantages and this PR is rather stalled now after this.

Reply:
Yes but that's how it works, Terminals interprets as they wish. Even colors can be represented as another one, I've seen yellow close to a brown.
`Bright` or not, the text is nearly always at least `Bold` - wider characters.

Reply:
Well not for me for example:
![image](https://user-images.githubusercontent.com/35064754/57638607-6de51600-75ae-11e9-9baf-46e43e0fa289.png)

Reply:
After looking at other libraries, the consensus seems to be `bold`, also on par with the VT100 documentation:
- https://github.com/fazibear/colorize/blob/master/lib/colorize/class_methods.rb#L78
- https://github.com/logrusorgru/aurora/blob/master/color.go#L60
- https://github.com/ogham/rust-ansi-term/blob/master/src/ansi.rs#L35

I can see issues opened by users saying they set bright and it does bold, or vice-versa, and this doesn't work. Look at this questions:
- [gnome-terminal use bright colors for bold](https://askubuntu.com/questions/875102/gnome-terminal-use-bright-colors-for-bold)
- [How do I make XTerm not use bold?](https://superuser.com/questions/15226/how-do-i-make-xterm-not-use-bold)
- [Disable Using Bright Colors for Bold Font in gnome-terminal](https://superuser.com/questions/516123/disable-using-bright-colors-for-bold-font-in-gnome-terminal)

I suggest having at least a description:
> Makes the text bold **or** bright, depending the terminal

Reply:
After looking at other libraries, the consensus seems to be `bold`, also on par with the VT100 documentation:
- https://github.com/fazibear/colorize/blob/master/lib/colorize/class_methods.rb#L78
- https://github.com/logrusorgru/aurora/blob/master/color.go#L60
- https://github.com/ogham/rust-ansi-term/blob/master/src/ansi.rs#L35

I can see issues opened by users saying they set bright and it does bold, or vice-versa, and this doesn't work. Look at this questions:
- [gnome-terminal use bright colors for bold](https://askubuntu.com/questions/875102/gnome-terminal-use-bright-colors-for-bold)
- [How do I make XTerm not use bold?](https://superuser.com/questions/15226/how-do-i-make-xterm-not-use-bold)
- [Disable Using Bright Colors for Bold Font in gnome-terminal](https://superuser.com/questions/516123/disable-using-bright-colors-for-bold-font-in-gnome-terminal)

I suggest having at least a description:
> Makes the text bold and/or bright, depending the terminal

Reply:
Actually, yeah I think it wouldn't be so bad to remove `Mode::Bright` because "bold" can really mean both "bright" and "bold" because sometimes "bold" is also used in other cases like "you made a bold statement there" which means you made like a "powerful"/"brave"/"big" statement and these words are in some way similar to "bright".

Reply:
Actually, it would be better to deprecate this method at least because it's used sometimes: https://github.com/search?l=Crystal&q=bright&type=Code.